torch/_inductor/triton_heuristics.py

```diff
@@ -53,6 +53,45 @@ class HeuristicType(Enum):
     TEMPLATE = auto()
 
 
+class AutotuneHint(Enum):
+    ELEMENTS_PER_WARP_32 = 0
+
+    # Triton codegen tries to codegen set of AutotuneHints.
+    # Enum.__repr__ looks like "<AutotuneHint.ELEMENTS_PER_WARP_32: 0>""
+    # which isn't valid python.
+    # Enum.__str__ will just return "AutotuneHint.ELEMENTS_PER_WARP_32".
+    __repr__ = Enum.__str__
+
+
+def autotune_hints_to_configs(
+    hints: Set[AutotuneHint], size_hints, block_size
+) -> List[Config]:
+    configs = []
+
+    for hint in hints:
+        if hint == AutotuneHint.ELEMENTS_PER_WARP_32:
+            if len(size_hints) == 1:
+                xyz_options = [(block_size // 4,)]
+            elif len(size_hints) == 2:
+                xyz_options = ((block_size // 4, 1), (1, block_size // 4))
+            elif len(size_hints) == 3:
+                xyz_options = (
+                    (block_size // 4, 1, 1),
+                    (1, block_size // 4, 1),
+                    (1, 1, block_size // 4),
+                )
+            for xyz in xyz_options:
+                configs.append(
```

Comment:
This is adding one config per dimension where previously you just added one. Is that intentional ? needed ?

Reply:
Intentional, yes; needed, idk.

Reply:
did some experiments, it seems like it _is_ useful to have multiple configs (e.g. (1, 256) and (256, 1)) - the best option will sometimes depend on the shape of size_hints